backend/controllers/auth.js

```diff
@@ -0,0 +1,135 @@
+const crypto = require("crypto");
+const ErrorResponse = require("../utils/errorResponse");
+const User = require("../models/User");
+const sendEmail = require("../utils/sendEmail");
+
+
+exports.login = async (req, res, next) => {
+  const { email, password } = req.body;
+
+
+  if (!email || !password) {
+    return next(new ErrorResponse("Please provide an email and password", 400));
+  }
+
+  try {
+    const user = await User.findOne({ email }).select("+password");
+
+    if (!user) {
+      return next(new ErrorResponse("Invalid credentials", 401));
+    }
+    const isMatch = await user.matchPassword(password);
+
+    if (!isMatch) {
+      return next(new ErrorResponse("Invalid credentials", 401));
+    }
+
+    sendToken(user, 200, res);
+  } catch (err) {
+    next(err);
+  }
+};
+
+
+exports.register = async (req, res, next) => {
+  const { username, email, password } = req.body;
+
+  try {
+    const user = await User.create({
+      username,
+      email,
+      password,
+    });
+
+    sendToken(user, 200, res);
+  } catch (err) {
+    next(err);
+  }
+};
+
+
+exports.forgotPassword = async (req, res, next) => {
+
+  const { email } = req.body;
+
+  try {
+    const user = await User.findOne({ email });
+
+    if (!user) {
+      return next(new ErrorResponse("No email could not be sent", 404));
+    }
+
+    const resetToken = user.getResetPasswordToken();
+
+    await user.save();
+
+    const resetUrl = `http://localhost:5173/passwordreset/${resetToken}`;
+
+
```

Comment:
We can use a .env var for the url. This will only work for localhost (i.e dev env). Why is the port 5173?